src/main/java/com/scaleunlimited/flinkcrawler/tools/CrawlTopology.java

```diff
@@ -186,48 +192,51 @@ public CrawlTopologyBuilder setJobName(String jobName) {
         }
 
         public CrawlTopologyBuilder setForceCrawlDelay(long forceCrawlDelay) {
-        	_forceCrawlDelay = forceCrawlDelay;
+            _forceCrawlDelay = forceCrawlDelay;
             return this;
         }
 
         public CrawlTopologyBuilder setDefaultCrawlDelay(long defaultCrawlDelay) {
-        	_defaultCrawlDelay = defaultCrawlDelay;
+            _defaultCrawlDelay = defaultCrawlDelay;
             return this;
         }
 
         public CrawlTopologyBuilder setUrlSource(BaseUrlSource urlSource) {
             _urlSource = urlSource;
             return this;
         }
-        
-        public CrawlTopologyBuilder setUrlLengthener(BaseUrlLengthener lengthener) {
+
+        public CrawlTopologyBuilder setUrlLengthener(SimpleUrlLengthener lengthener) {
```

Comment:
Why aren't we using `BaseUrlLengthener` here?

Reply:
It's because I didn't make setUserAgent an abstract method of BaseUrlLengthener. Would you go that far? Would you move _fetcher in there as well?

Reply:
I wouldn't add `setUserAgent` to `BaseUrlLengthener`, no. But I would assume that `SimpleUrlLengthener` has an http builder inside it, which is set up with the user agent name.

Reply:
LengthenUrlsFunction needs more than just a BaseHttpFetcherBuilder in order to build its _lengthener object, so it must either be constructed with a pre-built BaseUrlLengthener (as I'm doing now), or it must know itself to construct a SimpleUrlLengthener (unless passed a pre-built lengthener?), or I have to define a BaseUrlLengthenerBuilder class. I imagine you'd suggest that last option, but this does seem like a lot of extra baggage to me.

Reply:
I think the way we'd handled this elsewhere was for the function's `open()` method (`LengthenUrlsFunction.open()` in this case) to call an `open()` method for the object it's using under the hood (`BaseUrlLengthener` in this case). That way the wrapped object can create whatever non-serializable instance variables it needs.

Reply:
Done (by having the BaseUrlLengthener constructed if necessary by CrawlTopologyBuilder.build instead of at the top of the CrawlTopologyBuilder)